In [3]:
import numpy as np 
import pandas as pd 

In [4]:
data_challenge_path = "../data/raw/DATASET_CHALLENGE.csv"
dataset = pd.read_csv(data_challenge_path)

In [5]:
dataset.head(5)

codigo_cliente  edad     trabajo estado_civil credito_en_mora  \
0  H9BxQUn2k9SUE  29.0  estudiante      soltero              no   
1  dReetG5NK0ixs  34.0      obrero       casado              no   
2  NKM7y8wyr7yo2  45.0      obrero       casado              no   
3  PdS1UNNpxg1Gz  50.0      obrero       casado              no   
4  KY3dAInw6STSu  39.0     tecnico          NaN              no   

  credito_hipoteca credito_personal contacto  mes dia_semana  ... campania  \
0               no               no  celular  nov        jue  ...      3.0   
1               si               no  celular  nov        jue  ...      1.0   
2               si               no  celular  may        mar  ...      2.0   
3               si               no  celular  jul        lun  ...      2.0   
4               no               no  celular  jul        jue  ...      6.0   

   dias_transcurridos  contactos_previos  resultado_previas tasa_empleo  \
0                 3.0                2.0              exito        -3.4   
1               999.0                0.0        inexistente        -0.1   
2               999.0                0.0        inexistente        -1.8   
3               999.0                0.0        inexistente         1.4   
4               999.0                0.0        inexistente         1.4   

   indice_precios_consum  indice_confianza_consum  tasa_credito_hipotecario  \
0                 92.649                    -30.1                     0.714   
1                 93.200                    -42.0                     4.076   
2                 92.893                    -46.2                     1.344   
3                 93.918                    -42.7                     4.962   
4                 93.918                    -42.7                     4.968   

   personas_empleadas  resultado  
0              5017.5         si  
1              5195.8         no  
2              5099.1         si  
3              5228.1         si  
4              5228.1         no  

[5 rows x 21 columns]

In [6]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3850 entries, 0 to 3849
Data columns (total 21 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   codigo_cliente            3850 non-null   object 
 1   edad                      3847 non-null   float64
 2   trabajo                   3847 non-null   object 
 3   estado_civil              3845 non-null   object 
 4   credito_en_mora           3845 non-null   object 
 5   credito_hipoteca          3849 non-null   object 
 6   credito_personal          3847 non-null   object 
 7   contacto                  3814 non-null   object 
 8   mes                       3848 non-null   object 
 9   dia_semana                3846 non-null   object 
 10  educacion                 3841 non-null   object 
 11  campania                  3850 non-null   float64
 12  dias_transcurridos        3850 non-null   float64
 13  contactos_previos         3850 non-null   float64
 14  resultad

In [7]:
df_wo_nulls = dataset.dropna()
df_wo_nulls.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3778 entries, 0 to 3849
Data columns (total 21 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   codigo_cliente            3778 non-null   object 
 1   edad                      3778 non-null   float64
 2   trabajo                   3778 non-null   object 
 3   estado_civil              3778 non-null   object 
 4   credito_en_mora           3778 non-null   object 
 5   credito_hipoteca          3778 non-null   object 
 6   credito_personal          3778 non-null   object 
 7   contacto                  3778 non-null   object 
 8   mes                       3778 non-null   object 
 9   dia_semana                3778 non-null   object 
 10  educacion                 3778 non-null   object 
 11  campania                  3778 non-null   float64
 12  dias_transcurridos        3778 non-null   float64
 13  contactos_previos         3778 non-null   float64
 14  resultado_pre

From 3850 entries to 3778, deleting 1.87% data.

## Analizing variables to use on the prediction

To make the report analysis we should use all variables, even we are trying to make a report on Power BI, data visualitation with all variables

In [9]:
df_wo_nulls.to_csv('../data/processed/cdts_analysis.csv', index=False)

### Which varaibles will be useful to predict?

In [11]:
df_wo_nulls['contacto'].unique()

array(['celular', 'tel_fijo'], dtype=object)

In [12]:
df_wo_nulls.drop(columns=['codigo_cliente'])

edad         trabajo estado_civil credito_en_mora credito_hipoteca  \
0     29.0      estudiante      soltero              no               no   
1     34.0          obrero       casado              no               si   
2     45.0          obrero       casado              no               si   
3     50.0          obrero       casado              no               si   
6     41.0          obrero   divorciado              no               no   
...    ...             ...          ...             ...              ...   
3844  50.0     desempleado       casado     desconocido               no   
3845  35.0  administrativo       casado              no      desconocido   
3847  41.0  administrativo      soltero              no      desconocido   
3848  53.0  administrativo       casado              no               si   
3849  30.0       servicios      soltero              no               no   

     credito_personal  contacto  mes dia_semana                  educacion  \
0                  no   celular  nov        jue        tecnico_profesional   
1                  no   celular  nov        jue              basica_9anios   
2                  no   celular  may        mar              basica_9anios   
3                  no   celular  jul        lun              basica_4anios   
6                  no   celular  ago        mar              basica_9anios   
...               ...       ...  ...        ...                        ...   
3844               no   celular  ago        vie        tecnico_profesional   
3845      desconocido  tel_fijo  jul        mar  profesional_universitario   
3847      desconocido   celular  abr        jue             basica_11anios   
3848               no   celular  ago        mie  profesional_universitario   
3849               no   celular  jul        jue             basica_11anios   

      campania  dias_transcurridos  contactos_previos resultado_previas  \
0          3.0                 3.0                2.0             exito   
1          1.0               999.0                0.0       inexistente   
2          2.0               999.0                0.0       inexistente   
3          2.0               999.0                0.0       inexistente   
6          1.0               999.0                0.0       inexistente   
...        ...                 ...                ...               ...   
3844       1.0               999.0                0.0       inexistente   
3845       4.0               999.0                0.0       inexistente   
3847       2.0               999.0                0.0       inexistente   
3848       2.0               999.0                0.0       inexistente   
3849       1.0               999.0                0.0       inexistente   

      tasa_empleo  indice_precios_consum  indice_confianza_consum  \
0            -3.4                 92.649                    -30.1   
1            -0.1                 93.200                    -42.0   
2            -1.8                 92.893                    -46.2   
3             1.4                 93.918                    -42.7   
6             1.4                 93.444                    -36.1   
...           ...                    ...                      ...   
3844          1.4                 93.444                    -36.1   
3845          1.4                 93.918                    -42.7   
3847         -1.8                 93.075                    -47.1   
3848          1.4                 93.444                    -36.1   
3849          1.4                 93.918                    -42.7   

      tasa_credito_hipotecario  personas_empleadas resultado  
0                        0.714              5017.5        si  
1                        4.076              5195.8        no  
2                        1.344              5099.1        si  
3                        4.962              5228.1        si  
6                        4.963              5228.1        no  
...                        ...                 .

In [13]:
df_wo_nulls.to_csv('../data/processed/cdts_prediction.csv', index=False)